core/src/env.rs

```diff
@@ -108,8 +108,11 @@ pub trait Ext {
     /// Access currently handled message payload.
     fn msg(&mut self) -> &[u8];
 
-    /// Report that some gas has been used.
-    fn gas(&mut self, amount: u32) -> Result<(), &'static str>;
+    /// Charge some gas. TODO: why u32?
```

Comment:
`u32` as this function was called by the instrumented function which consumes not too much gas per weight unit. 

We use the value of `1000` at the moment: https://github.com/gear-tech/gear/blob/master/core/src/gas.rs#L169

Reply:
It's very strange decision. Because in config:  https://github.com/gear-tech/gear/blob/94a8ee76f1b62d513f4a0b570f854b68d2568927/core-runner/src/runner.rs#L54
we have u64 costs, and u64 -> u32 cast may be dangerous

Reply:
If you intend to use this function not only by injected gas counter you may change the signature and convert `u32` to `u64` in correspondent calls.

Reply:
Sorry, but I didn't understand anything at all =)
I want to use this function only for gas counters.
May change signature of what?
As I see the only one impl of this trait is in core-runner. And it uses u64 gas counter. So, I think the most safe way is use the same types for counter and for amount which we want to charge or refund.

Reply:
@NikVolf, can you comment this?

Reply:
Any single gas charge from inside the wasm is u32

Reply:
Ok, that a good argument. Can you please tell me how can I find where in gear this function is used? https://github.com/gear-tech/gear/blob/94a8ee76f1b62d513f4a0b570f854b68d2568927/core-backend/wasmtime/src/env.rs#L55

Reply:
When you inject the gas counter into the WASM program (see the link above), the call to `gas` function is added between WASM instructions. Our embedder exports this function to the WASM environment for the program could call it (in your quote of the code). The signature of this function is dictated by the injector (`pwasm_utils::inject_gas_counter`) and is `fn gas(value: u32)`. Of course, you can call your embedded function with `u64` gas value by converting the source gas value from `u32` to `u64`. Alternatively, you may distinguish between small gas values (e.g. fees per weight unit or per alloc) and big ones (gas limit etc.) and use correspondent types (u32 and u64).

Reply:
Ok thanks